CMakeLists.txt

```diff
@@ -132,16 +137,9 @@ elseif(CMAKE_C_COMPILER_ID MATCHES "GNU")
 elseif(CMAKE_C_COMPILER_ID MATCHES "MSVC")
     message(STATUS "libavif: Enabling warnings for MSVC")
     add_definitions(
-        /Wall # All warnings
-        /wd4255 # Disable: no function prototype given
+        /W4
```

Comment:
@joedrago FYI

`/Wall` generates a lot of warnings in GoogleTest headers and some warnings in our C++ tests. We could suppress the warnings in GoogleTest headers, and try to fix as many warnings in our C++ tests as we can and suppress the rest. But I decided to use `/W4` instead, which is recommended by this Visual Studio doc page:

https://docs.microsoft.com/en-us/cpp/build/reference/compiler-option-warning-level?view=msvc-170

> **/W4** displays level 1, level 2, and level 3 warnings, and all level 4 (informational) warnings that aren't off by default. We recommend that you use this option to provide lint-like warnings. For a new project, it may be best to use **/W4** in all compilations. This option helps ensure the fewest possible hard-to-find code defects.